src/test/java/edu/ucla/library/iiif/auth/services/AccessCookieServiceTest.java

```diff
@@ -203,7 +234,39 @@ public final void testValidateSinaiCookie(final Vertx aVertx, final VertxTestCon
         }).onFailure(aContext::failNow);
     }
 
+    /**
+     * Tests that an expired Sinai cookie will not be accepted.
+     *
+     * @param aVertx A Vert.x instance
+     * @param aContext A test context
+     */
+    @Test
+    public final void testInvalidateExpiredSinaiCookie(final Vertx aVertx, final VertxTestContext aContext) {
+        final Future<Boolean> validateCookie4DaysOld = validateSinaiCookieTuple(myMockSinaiCookies4DaysOld);
+
+        validateCookie4DaysOld.onSuccess(result -> {
+            try {
+                assertFalse(result);
+
+                aContext.completeNow();
```

Comment:
No need to see if it's completed before completeNow? Does this no longer output the warning if you complete something that's already completed (because it's failed the assertion)?

Reply:
I don't think it's possible that it can be completed if the assertion fails, since control then goes to the catch block. I was honestly never sure why our other projects had that issue. Maybe that was a bug in the older Vert.x?

Reply:
We were definitely not seeing it end at the assertion check. It failed the context but kept executing until the end of the block (didn't throw the exception until then). Not sure if that was a bug or expected behavior.

Reply:
I just ran this test locally with `assertFalse(!result)` instead. Confirming that I do not see the "context was already completed" message that we were seeing with the other projects. 🤷🏻

Reply:
Interesting, and good to know. Thanks!